lib/pbench/test/unit/server/test_datasets_daterange.py

```diff
@@ -0,0 +1,134 @@
+import datetime
+from http import HTTPStatus
+import requests
+from typing import Dict, List
+
+import pytest
+
+from pbench.server import PbenchServerConfig, JSON
+from pbench.server.database.models.datasets import Dataset
+
+
+class TestDatasetsDateRange:
+    """
+    Test the `datasets/daterange` API. We perform a variety of queries using a
+    set of datasets provided by the `attach_dataset` fixture and the
+    `more_datasets` fixture.
+    """
+
+    @pytest.fixture()
+    def query_as(self, client, server_config, more_datasets, provide_metadata):
+        """
+        Helper fixture to perform the API query and validate an expected
+        return status.
+
+        Args:
+            client: Flask test API client fixture
+            server_config: Pbench config fixture
+            more_datasets: Dataset construction fixture
+            provide_metadata: Dataset metadata fixture
+        """
+
+        def query_api(
+            payload: JSON, username: str, expected_status: HTTPStatus
+        ) -> requests.Response:
+            token = self.token(client, server_config, username)
+            response = client.get(
+                f"{server_config.rest_uri}/datasets/daterange",
+                headers={"authorization": f"bearer {token}"},
+                query_string=payload,
+            )
+            assert response.status_code == expected_status
+            return response
+
+        return query_api
+
+    def token(self, client, config: PbenchServerConfig, user: str) -> str:
+        response = client.post(
+            f"{config.rest_uri}/login",
+            json={"username": user, "password": "12345"},
+        )
+        assert response.status_code == HTTPStatus.OK
+        data = response.json
+        assert data["auth_token"]
+        return data["auth_token"]
+
+    def get_results(self, name_list: List[str]) -> Dict[str, datetime.datetime]:
+        """
+        Use a list of "expected results" to determine the earliest and the
+        latest creation date of the set of datasets.
+
+        Args:
+            name_list: List of dataset names
+
+        Returns:
+            {"from": first_date, "to": last_date}
+        """
+        from_time = datetime.datetime.now(datetime.timezone.utc)
```

Comment:
Given our propensity for playing with freezeguns, I would suggest setting this to a known value, e.g.:
```
        to_time = datetime.datetime(
            year=9999, month=12, day=31, tzinfo=datetime.timezone.utc
        )
```

Reply:
Why, because, in the future, we might create a dataset in the future? 🤣 

If we are going to worry about that, I'm not sure there's any good fixed upper bound. I mean, we might want to make sure that Pbench still works in the year 10000... we'll probably have a managed service by then.

(Sigh.) I suppose I'll just go back to my first rejected gut instinct, which was to start them at `None` and assign `to_time` and `from_time` from the first dataset in the list, then work from there. I thought this was simpler.

Reply:
My concern is that, given the way we use freezegun, `now()` might evaluate to an inconvenient value (but, perhaps it will be no newer than the datasets' timestamps 😆).